src/shared/Microsoft.Git.CredentialManager/IGit.cs

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+using System;
+
+namespace Microsoft.Git.CredentialManager
+{
+    public interface IGitConfiguration : IDisposable
+    {
+        /// <summary>
+        /// Path to repository if this configuration object is also scoped to a repository, otherwise null.
+        /// </summary>
+        string RepositoryPath { get; }
+
+        /// <summary>
+        /// Get the value of a configuration entry as a string, or null if not found.
+        /// Equivalent to <seealso cref="GetString"/>.
```

Comment:
If this is an equivalent, then can we make one of them an extension method so we don't need to duplicate the redirection in every implementation?

Reply:
Since you already have extension methods below, perhaps `.GetString(string name)` can be in there, too.

Reply:
I've changed the design here. Instead we now have a `TryGetString` method which is on the interface, and extension methods for `GetString` (and the overloads).